libs/common/src/platform/services/crypto.service.spec.ts

```diff
@@ -340,4 +348,327 @@ describe("cryptoService", () => {
+      makeMasterKey,
+      makeUserKey,
+    }: {
+      makeMasterKey: boolean;
+      makeUserKey: boolean;
+    }): [UserKey, MasterKey] => {
+      const userKeyState = stateProvider.singleUser.getFake(mockUserId, USER_KEY);
+      const fakeMasterKey = makeMasterKey ? makeSymmetricCryptoKey<MasterKey>(64) : null;
+      masterPasswordService.masterKeySubject.next(fakeMasterKey);
+      userKeyState.stateSubject.next([mockUserId, null]);
+      const fakeUserKey = makeUserKey ? makeSymmetricCryptoKey<UserKey>(64) : null;
+      userKeyState.stateSubject.next([mockUserId, fakeUserKey]);
+      return [fakeUserKey, fakeMasterKey];
+    };
+
+    it("will return users decrypted private key when legacy support set to %legacySupport and user hasUserKey = $hasUserKey and user hasMasterKey = $hasMasterKey", async () => {
+      const [userKey] = setupKeys({
+        makeMasterKey: false,
+        makeUserKey: true,
+      });
+
+      const userEncryptedPrivateKeyState = stateProvider.singleUser.getFake(
+        mockUserId,
+        USER_ENCRYPTED_PRIVATE_KEY,
+      );
+
+      const fakeEncryptedUserPrivateKey = makeEncString("1");
+
+      userEncryptedPrivateKeyState.stateSubject.next([
+        mockUserId,
+        fakeEncryptedUserPrivateKey.encryptedString,
+      ]);
+
+      // Decryption of the user private key
+      const fakeDecryptedUserPrivateKey = makeStaticByteArray(10, 1);
+      encryptService.decryptToBytes.mockResolvedValue(fakeDecryptedUserPrivateKey);
+
+      const fakeUserPublicKey = makeStaticByteArray(10, 2);
+      cryptoFunctionService.rsaExtractPublicKey.mockResolvedValue(fakeUserPublicKey);
+
+      const userPrivateKey = await firstValueFrom(cryptoService.userPrivateKey$(mockUserId));
+
+      expect(encryptService.decryptToBytes).toHaveBeenCalledWith(
+        fakeEncryptedUserPrivateKey,
+        userKey,
+      );
+
+      expect(userPrivateKey).toBe(fakeDecryptedUserPrivateKey);
+    });
+
+    it("returns null user private key when no user key is found", async () => {
+      setupKeys({ makeMasterKey: false, makeUserKey: false });
+
+      const userPrivateKey = await firstValueFrom(cryptoService.userPrivateKey$(mockUserId));
+
+      expect(encryptService.decryptToBytes).not.toHaveBeenCalled();
+
+      expect(userPrivateKey).toBeFalsy();
+    });
+
+    it("returns null when user does not have a private key set", async () => {
+      setupKeys({ makeUserKey: true, makeMasterKey: false });
+
+      const encryptedUserPrivateKeyState = stateProvider.singleUser.getFake(
+        mockUserId,
+        USER_ENCRYPTED_PRIVATE_KEY,
+      );
+      encryptedUserPrivateKeyState.stateSubject.next([mockUserId, null]);
+
+      const userPrivateKey = await firstValueFrom(cryptoService.userPrivateKey$(mockUserId));
+      expect(userPrivateKey).toBeFalsy();
+    });
+  });
+
+  describe("cipherDecryptionKeys$", () => {
+    const fakePrivateKeyDecryption = (encryptedPrivateKey: Encrypted, key: SymmetricCryptoKey) => {
```

Comment:
nit: Same here, write it as a function

```suggestion
    function fakePrivateKeyDecryption(encryptedPrivateKey: Encrypted, key: SymmetricCryptoKey) {
```

Reply:
Changed to function